storage/remote/queue_manager.go

```diff
@@ -398,6 +443,37 @@ func (t *QueueManager) Stop() {
 	t.metrics.unregister()
 }
 
+// Getting the data from a shard and structuring it as a SegmentRecord from a shard
+func (t *QueueManager) getSegmentRecords() SegmentRecord {
+	RecordSegment := t.shards.qm.watcher.SegmentFile
+	RecordEndpoint := t.shards.qm.url
+
+	record := SegmentRecord{
+		Segment:  RecordSegment,
+		Endpoint: RecordEndpoint,
+	}
+
+	return record
+}
+
+// Writes the segment record in a JSON file
+func (t *QueueManager) RecordSegment() {
+	Checkpoint.Recorded = time.Now()
+	Checkpoint.Checkpoints = append(Checkpoint.Checkpoints, t.getSegmentRecords())
+
+	// now Marshal it
+	data, err := json.MarshalIndent(Checkpoint, "", "")
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	err = ioutil.WriteFile("SegmentRecord.json", data, 0644)
```

Comment:
Have a look at how we create and write the normal WAL checkpoints: https://github.com/prometheus/prometheus/blob/master/tsdb/wal/checkpoint.go#L84-L257 Note the use of a temporary dir/file. What would happen if we, for some reason, were reading the current remote write checkpoint file and `RecordSegment` was called.